File: ejercicio4/FilterManager.java
package ejercicio4;

class FilterManager {
    private FilterChain filterChain;

    public FilterManager(AuthTarget target) {
        filterChain = new FilterChain();
        filterChain.setTarget(target);
    }

    public void addFilter(Filter filter) {
        filterChain.addFilter(filter);
    }

    public void authenticate(Message message) throws Exception {
        filterChain.execute(message);
    }
}

File: P2/email_filter_app/lib/filter_manager.dart
import 'package:email_filter_app/credentials.dart';
import 'package:email_filter_app/auth_target.dart';
import 'package:email_filter_app/filters/filter.dart';
import 'package:email_filter_app/filters/filter_chain.dart';

class FilterManager {
  final FilterChain _filterChain;

  FilterManager(AuthTarget target) : _filterChain = FilterChain() {
    _filterChain.setTarget(target);
  }

  void addFilter(Filter filter) {
    _filterChain.addFilter(filter);
  }

  void authenticate(Credentials credentials) {
    _filterChain.execute(credentials);
  }
}
